NDTensors/src/abstractarray/tensoralgebra/contract.jl

```diff
@@ -162,13 +168,21 @@ function _contract!(
 
   #tC = similar(CM)
   #_gemm!(tA, tB, El(α), AM, BM, El(β), CM)
-  CM = mul!!(CM, AM, BM, El(α), El(β))
+  ## If C^T = A^T * B^T => C = B * A
+  ## TODO There is an issue in CUDA.jl
+  ## When all are transpose CUDA.mul! isn't being
+  ## Called correctly in `NDTensorsCUDAExt`
+  if AM isa Transpose && BM isa Transpose && CM isa Transpose
+    CM = mul!!(parent(CM), parent(BM), parent(AM), El(α), El(β))
+  else
+    CM = mul!!(CM, AM, BM, El(α), El(β))
+  end
```

Comment:
Seems like this should use `expose`.

Reply:
This is the logic I made before with `permutedims!!` the `mul!!` calls `mul!(expose(CM), expose(AM), expose(BM), a, b)`

Reply:
So maybe this should be handled with normal dispatch, i.e. define:
```julia
function mul!!(C::Transpose, A::Transpose, B::Transpose, α, β)
  CT = mul!!(transpose(C), transpose(B), transpose(A), α, β)
  return transpose(CT)
end
```
Note that we want to make sure it outputs the correctly transposed version. The point is more that this kind of logic should be handled outside of a high level function like `contract!`.

As a guiding principle, it is usually better to avoid code like `if A isa T ...` and instead use function dispatch.

Reply:
This makes a lot of sense, thanks! Also I am not sure that I need to return a transpose of CT because 
```
C(k,i)^T = A(j,i)^T B(k,j)^T => C(k,i) = B(k,j) A(j,i)
```